### PR TITLE
Fix data races, found on OSX

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by eile** [comment]
+1

File: brain/detail/circuit.h [hunk 1]
```diff
@@ -310,10 +322,13 @@ class Circuit::Impl
...
-    mutable std::unique_ptr< brion::Synapse > _synapseAttributes[2];
-    mutable std::unique_ptr< brion::Synapse > _synapseExtra;
-    mutable std::unique_ptr< brion::Synapse > _synapsePositions[2];
+    template< typename T >
+    using LockPtr = lunchbox::Lockable< std::unique_ptr< T >>;
+
+    mutable Ptr< brion::SynapseSummary > _synapseSummary;
+    mutable Ptr< brion::Synapse > _synapseAttributes[2];
+    mutable Ptr< brion::Synapse > _synapseExtra;
+    mutable Ptr< brion::Synapse > _synapsePositions[2];
```
Comment (hernando) [inline brain/detail/circuit.h]: LockPtr

Comment (tribal-tec) [inline brain/detail/circuit.h]: Indeed! Seems I'm still asleep...

**Comment by hernando** [comment]
Rebase on top of master, that should solve the last compilation error.

**Pull request merged**